### PR TITLE
Add giflib/5.2.1

### PR DESCRIPTION
**Pull request by p-podsiadly**
**giflib/5.2.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

This PR adds the latest version of giflib. Version 5.1.4, which is the only version in CCI is 4 years old.

In the meantime, upstream removed Autotools build system, the library now has only Makefile for Unix. For this reason, I had to add custom `CMakeLists.txt`.

### CODE REVIEW

**Comment by p-podsiadly** [comment]
> 
> 
> @p-podsiadly Please, consider [p-podsiadly#1](https://github.com/p-podsiadly/conan-center-index/pull/1)

I applied your changes to this PR, thanks! I'm not sure about source patching - should it be done in `source` or `build` method?

File: recipes/giflib/5.2.x/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("%s-%s" % (self.name, self.version), self._source_subfolder)
+        
+        self._patch_sources()
```
Comment (uilianries) [inline recipes/giflib/5.2.x/conanfile.py:34]: Move to build folder. Keep source folder untouched. You can change/remove the build folder easily, but build folder requires a new download.

File: recipes/giflib/5.2.x/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import os
+
+class GiflibConan(ConanFile):
+    name = "giflib"
+    description = "A library and utilities for reading and writing GIF images."
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    homepage = "http://giflib.sourceforge.net"
+    topics = ("conan", "giflib", "image", "multimedia", "format", "graphics")
+    settings = "os", "arch", "compiler", "build_type"
```
Comment (uilianries) [inline recipes/giflib/5.2.x/conanfile.py:12]:
```suggestion
    settings = "os", "arch", "compiler", "build_type"
    options = {"shared": [True, False], "fPIC": [True, False]}
    default_options = {"shared": False, "fPIC": True}
```

File: recipes/giflib/5.2.x/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
```
Comment (uilianries) [inline recipes/giflib/5.2.x/conanfile.py:22]:
```suggestion
    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC

    def configure(self):
        if self.options.shared:
            del self.options.fPIC
```

**Pull request merged**